### PR TITLE
Add `ILikeHandler`

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Reader/FilterHandler/ILikeHandler/ILikeHandlerFactory.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace Yiisoft\Data\Cycle\Reader\FilterHandler\ILikeHandler;
+
+use Cycle\Database\Driver\DriverInterface;
+use RuntimeException;
+use Yiisoft\Data\Reader\FilterHandlerInterface;
+
+class ILikeHandlerFactory
+{
+    public static function getIlikeHandler(?DriverInterface $databaseDriver): FilterHandlerInterface
```
Comment (vjik) [inline src/Reader/FilterHandler/ILikeHandler/ILikeHandlerFactory.php:13]: This method used in `EntityReader` only, maybe move it there as private?

File: src/Reader/FilterHandler/ILikeHandler/ILikeHandlerFactory.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        $driverType = $databaseDriver?->getType() ?? 'SQLite';
```
Comment (vjik) [inline src/Reader/FilterHandler/ILikeHandler/ILikeHandlerFactory.php:15]: Is there case when database driver not exist?

File: tests/Feature/Base/Reader/FilterHandler/LikeHandlerTest.php [hunk 3]
```diff
@@ -4,15 +4,25 @@
...
 abstract class LikeHandlerTest extends BaseData
 {
-    public function testLikeHandler(): void
+    public static function dataBase(): array
     {
-        $reader = (new EntityReader($this->select('user')))->withFilter(new Like('email', 'seed@%'));
-        $this->assertFixtures([2], $reader->read());
+        return [
+            'case matches' => ['email', 'seed@%', [2]],
+            'case does not match' => ['email', 'SEED@%', [2]],
```
Comment (vjik) [inline tests/Feature/Base/Reader/FilterHandler/LikeHandlerTest.php:18]: Should `like` always be case sensitive?

**Review by vjik — CHANGES_REQUESTED** [review]

**Comment by arogachev** [comment]
Replaced by #31.

**Pull request closed without merging**